### PR TITLE
AE-4245 Fixes+tests

### PR DESCRIPTION
**Pull request by pgan002**
https://iamplus.atlassian.net/browse/AE-4245

### CODE REVIEW

File: timestring/Range.py [hunk 1]
```diff
@@ -33,25 +33,23 @@ def __init__(self, start, end=None, offset=None, start_of_week=0, tz=None, verbo
...
 
         elif start == 'infinity':
-            # end was not provided
+            # End was not provided
             self._dates = (Date('infinity'), Date('infinity'))
 
         elif re.search(r'(\s(and|to)\s)', start):
-            """Both sides where provided in the start
-            """
+            # Both sides aer provided in string "start"
```
Comment (dpathania1) [inline timestring/Range.py]: typo

File: timestring/Range.py [hunk 2]
```diff
@@ -33,25 +33,23 @@ def __init__(self, start, end=None, offset=None, start_of_week=0, tz=None, verbo
...
             start = re.sub('^(between|from)\s', '', start.lower())
             # Both arguments found in start variable
             r = tuple(re.split(r'(\s(and|to)\s)', start.strip()))
-            self._dates = (Date(r[0], tz=tz), Date(r[-1], tz=tz))
+            start = Range(r[0], tz=tz).start
```
Comment (dpathania1) [inline timestring/Range.py:48]: what is the use-case for this change?

File: timestring/Range.py [hunk 3]
```diff
@@ -169,14 +180,41 @@ def __init__(self, start, end=None, offset=None, start_of_week=0, tz=None, verbo
...
+                    if second:
+                        end = start + '1 second'
+                    elif minute:
+                        end = start + '1 minute'
+                    elif hour:
+                        end = start + '1 hour'
+                    elif day:
+                        end = start + '1 day'
+                    elif month:
+                        end = start + '1 month'
```
Comment (dpathania1) [inline timestring/Range.py]: Is it not 1 day for month and year?

**Review by dpathania1 — APPROVED** [review]

File: timestring/Range.py (hunk 2 above)
Comment (pgan002) [reply timestring/Range.py:48]: Month names and week days parsed by Date do not start on midnight, while we want them to start on midnight in a date range. Maybe we should change `Date`.

File: timestring/Range.py [hunk 4]
```diff
@@ -169,14 +180,41 @@ def __init__(self, start, end=None, offset=None, start_of_week=0, tz=None, verbo
...
+                        start = start.replace(second=0)
+
+                    if second:
+                        end = start + '1 second'
+                    elif minute:
+                        end = start + '1 minute'
+                    elif hour:
+                        end = start + '1 hour'
+                    elif day:
+                        end = start + '1 day'
```
Comment (pgan002) [inline timestring/Range.py:211]: By mistake I deleted @dpathania1 's comment that I was replying to. The comment asked shouldn't years ago be 1 day long. That change is in https://github.com/iAmPlus/timestring/pull/18, but I can put both of them in one PR if that's easier to review.

File: timestring/Range.py (hunk 1 above)
Comment (pgan002) [reply timestring/Range.py]: Thanks!

**Pull request merged**